packages/in_app_purchase/in_app_purchase_storekit/lib/src/store_kit_wrappers/enum_converters.dart

```diff
@@ -117,3 +117,27 @@ class _SerializedEnums {
   late SKSubscriptionPeriodUnit unit;
   late SKProductDiscountPaymentMode discountPaymentMode;
 }
+
+/// Serializer for [SKProductDiscountType].
+///
+/// Use these in `@JsonSerializable()` classes by annotating them with
+/// `@SKProductDiscountTypeConverter()`.
+class SKProductDiscountTypeConverter
+    implements JsonConverter<SKProductDiscountType, int?> {
+  /// Default const constructor.
+  const SKProductDiscountTypeConverter();
+
+  @override
+  SKProductDiscountType fromJson(int? json) {
+    if (json == null) {
+      return SKProductDiscountType.introductory;
+    }
+    return $enumDecode<SKProductDiscountType, dynamic>(
+        _$SKProductDiscountTypeEnumMap.cast<SKProductDiscountType, dynamic>(),
+        json);
```

Comment:
This method also qualifies to have additional unit tests. Especially the `null` check should be validates. It would be overkill to test all possible outcomes of the `$enumDecode` method however I do feel it is beneficial to test the following cases:

1. When `json` is `null`  the method should return `SKProductDiscountType.introductory`;
2. When `json` is not `null` the method should return a valid `SKProductDiscountType` value.

Reply:
Thanks for you reply. 
But I didn't find a similar test case in the previous code such as for `SKSubscriptionPeriodUnitConverter`.
Maybe it is not necessary I think?

Reply:
It is possible that this code was created before the requirement that all [PRs need tests](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests) was created. As per these guidelines this code needs a unit test.

Reply:
Done